files/common/usr/bin/delphix-startup-screen

```diff
@@ -0,0 +1,400 @@
+    each service's information is separated by an empty string (formatted
+    for clarity):
+    ['Delphix management service', 'active',
+    '',
+    'Delphix Masking Service', 'inactive']
+
+    Callers use the empty string to detect to detect information for a
+    new service.
+    """
+
+    fields: str = 'ActiveState'
+    cmd: List[str] = ['systemctl', 'show', '--value', '-p']
+    if options:
+        fields += ',' + options
+    cmd.append(fields)
+    for svc in get_svcs():
+        cmd.append(svc)
+    cp = subprocess.run(cmd,
+                        stdout=subprocess.PIPE,
+                        universal_newlines=True,
+                        check=True)
+    return cp.stdout.splitlines()
+
+
+def get_keyboard_layout() -> str:
+    """
+    Return the current keyboard layout.
+    """
+    cmd: List[str] = ['localectl', 'status']
+    cp = subprocess.run(cmd,
+                        stdout=subprocess.PIPE,
+                        universal_newlines=True,
+                        check=True)
+    output: List[str] = cp.stdout.split()
+    return output[output.index("Layout:") + 1]
+
+
+def set_keyboard_layout(layout: str) -> subprocess.CompletedProcess:
+    """
+    Set the keyboard layout based on the user selection.
+    """
+    cmd: List[str] = ['localectl', 'set-x11-keymap', layout, 'pc105']
+    subprocess.run(cmd, shell=False, check=True)
+    return subprocess.run('setupcon', shell=False, check=True)
+
+
+def get_valid_keyboard_layouts() -> List[str]:
+    """
+    Return valid keyboard layouts supported by the system.
+    """
+    cmd: List[str] = ['localectl', '--no-pager', 'list-x11-keymap-layouts']
+    cp = subprocess.run(cmd,
+                        stdout=subprocess.PIPE,
+                        universal_newlines=True,
+                        check=True)
+    return cp.stdout.splitlines()
+
+
+def load_header(stdscr) -> Any:
+    """
+    Display the header information for the main screen.
+    """
+    WIN_LEN, WIN_HT = set_common_variables(stdscr)
+    cmd = ['get-appliance-version']
+    cp = subprocess.run(cmd,
+                        stdout=subprocess.PIPE,
+                        universal_newlines=True,
+                        check=True)
+    version: str = cp.stdout
+
+    stdscr.clear()
+    stdscr.addstr(1, 2, LOGO + str(version), curses.A_BOLD)
+    stdscr.addstr(1, WIN_LEN - len(website) - 2, website, curses.A_BOLD)
+
+    win = stdscr.subwin(WIN_HT - 4, WIN_LEN - 4, 3, 2)
+    win.bkgd(' ', curses.color_pair(2))
+    win.box()
+    return win
+
+
+def is_server_down() -> bool:
+    """
+    Determine if the server is down by looking at the state of all the
+    services. If any service is 'failed' then we declare the server down.
+    """
+    for i in run_svcs():
+        #
+        # Empty strings separate services, skip those.
+        #
+        line: str = i.strip()
+        if line and line.strip() == "failed":
+            return True
+    return False
```

Comment:
[1] if `line` is equal to `i.strip()` then later when you do `line.strip()` it should be equal to `i.strip().strip()`. Is there a reason why we are stripping twice?
 
[2] In our previous conversation you said that the output that we iterate over is `['active', '', 'inactive']` and that's why you are checking if `line` is equal to the empty string, correct? My point before was that you can still compare the empty string to "failed" and continue to the next iteration of the loop:
```
$ python3
Python 3.6.9 (default, Nov  7 2019, 10:44:02)
[GCC 8.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> "" == "failed"
False
```
That said I understand if you want to keep it like that to warn the reader that `line` can also be an empty string, but I think the comment could do that for you.

Similarly for the `is_server_up()` function.

Reply:
I have reworked the functions so take another look.